impala/_thrift_api.py

```diff
@@ -120,6 +123,43 @@ def get_socket(host, port, use_ssl, ca_cert):
     else:
         return TSocket(host, port)
 
+def get_http_transport(host, port, http_path, timeout=None, use_ssl=False,
+                       ca_cert=None, auth_mechanism='NOSASL', user=None,
+                       password=None):
+    # TODO: support timeout
+    if timeout is not None:
+        log.error('get_http_transport does not support a timeout')
+    if use_ssl:
+        url = 'https://%s:%s/%s' % (host, port, http_path)
+        log.debug('get_http_transport url=%s', url)
+        # TODO: Add server authentication with thrift 0.12
```

Comment:
Should probably raise an exception if ca_cert was not provided and we're not checking it

Reply:
Done, I added the check to the calling code because I didn't want to introduce dbapi errors to the thrift api wrapper.